tests/unit/test_intrin.cpp

```diff
@@ -46,6 +52,76 @@ TEST(intrin, exp256) {
     }
 }
 
+TEST(intrin, abs256) {
+    constexpr size_t simd_len = 4;
+
+    __m256d vvalues[] = {
+        _mm256_set_pd(-42.,  -0.,  0.,  42.),
+        _mm256_set_pd(-dmin,  -dmax,  -dmax, dqnan),
```

Comment:
I think we should also test for `-Inf` and `Inf`.

Reply:
I am in two minds about this: sure for completeness, but on the other hand we never intend to have NaN or Inf in our models.